src/main/java/gyro/google/compute/RegionInstanceGroupManagerResource.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    /**
+     * The region where the managed instance group resides.
+     */
+    @Immutable
+    @Required
+    public String getRegion() {
+        return region;
+    }
+
+    public void setRegion(String region) {
+        this.region = region;
+    }
+
+    @Override
+    public boolean doRefresh() throws Exception {
+        Compute client = createClient(Compute.class);
+        copyFrom(client.regionInstanceGroupManagers().get(getProjectId(), getRegion(), getName()).execute());
+        return true;
+    }
+
+    @Override
+    public void doCreate(GyroUI ui, State state) throws Exception {
+        InstanceGroupManager instanceGroupManager = createInstanceGroupManager();
+        Optional.ofNullable(getDistributionPolicy())
+            .map(ComputeDistributionPolicy::copyTo)
+            .ifPresent(instanceGroupManager::setDistributionPolicy);
+        Compute client = createClient(Compute.class);
+        Operation operation = client.regionInstanceGroupManagers()
+            .insert(getProjectId(), getRegion(), instanceGroupManager)
+            .execute();
+        waitForCompletion(client, operation);
+        refresh();
+    }
+
+    @Override
+    public void doUpdate(GyroUI ui, State state, Resource current, Set<String> changedFieldNames)
+        throws Exception {
+        // TODO: target pool resource
+        // https://github.com/perfectsense/gyro-google-provider/issues/79
+        //    private List<AbstractTargetPoolResource> targetPools;
+        boolean shouldPatch = false;
+
+        InstanceGroupManager instanceGroupManager = new InstanceGroupManager();
+
+        for (String changedFieldName : changedFieldNames) {
+            // template changes
+            if (changedFieldName.equals("instance-template")) {
+                setInstanceTemplate();
+            } else if (changedFieldName.equals("target-size")) {
+                instanceGroupManager.setTargetSize(getTargetSize());
+                shouldPatch = true;
+            } else if (changedFieldName.equals("auto-healing-policy")) {
+                List<ComputeInstanceGroupManagerAutoHealingPolicy> diffableAutoHealingPolicy = getAutoHealingPolicy();
+                List<InstanceGroupManagerAutoHealingPolicy> autoHealingPolicies = null;
+
+                if (diffableAutoHealingPolicy.isEmpty()) {
+                    autoHealingPolicies = Data.nullOf(List.class);
+                } else {
+                    autoHealingPolicies = diffableAutoHealingPolicy
+                        .stream()
+                        .map(ComputeInstanceGroupManagerAutoHealingPolicy::copyTo)
+                        .collect(Collectors.toList());
+                }
+                instanceGroupManager.setAutoHealingPolicies(autoHealingPolicies);
+                shouldPatch = true;
+            } else if (changedFieldName.equals("named-port")) {
+                // TODO: investigate as GCP UI allows updating of named ports.
+            }
```

Comment:
If the api does not allowing the update of named ports then this should not be present.